pkg/apis/ceph.rook.io/v1/network_test.go

```diff
@@ -72,10 +72,54 @@ func TestValidateNetworkSpec(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+// test the NetworkSpec.IsHost method with different network providers
+func TestNetworkCephIsHost(t *testing.T) {
+	net := NetworkSpec{HostNetwork: false}
+
+	net.Provider = NetworkProviderHost
+	assert.True(t, net.IsHost())
+
+	net.Provider = NetworkProviderMultus
+	assert.False(t, net.IsHost())
+
+	// test with  nonempty but invalid provider
+	net.Provider = "foo"
+	assert.False(t, net.IsHost())
+
+	// test with  nonempty but invalid provider
+	net.Provider = "foo"
+	assert.False(t, net.IsHost())
+}
+
+// test the NetworkSpec.IsHost method with network providers in combination with the legacy
+// "HostNetwork  setting.
 func TestNetworkCephIsHostLegacy(t *testing.T) {
 	net := NetworkSpec{HostNetwork: true}
 
 	assert.True(t, net.IsHost())
+
+	net.Provider = NetworkProviderDefault
+	assert.True(t, net.IsHost())
+
+	net.Provider = NetworkProviderMultus
+	assert.False(t, net.IsHost())
+
+	net.HostNetwork = false
+	net.Provider = NetworkProviderDefault
+	assert.False(t, net.IsHost())
```

Comment:
@BlaineEXE  wrote:
> This test case I think belongs to the `TestNetworkCephIsHost` above based on my interpretation.

well, I have put this here because it tests an aspect of the interaction of Provider with the legacy HostNetwork setting.

Reply:
This is the only snippet in this test that has `net.HostNetwork = false` though. When `net.HostNetwork = false`, the legacy setting is no longer a consideration. I think this test should move up into the section where all the rest of the `net.HostNetwork = false` tests are.